src/mixins/atoms.js

```diff
@@ -71,18 +77,111 @@ export const AtomsMixin = (superclass) => class extends superclass {
     createAtomsGroup(basis, atomRadiiScale) {
         const atomsGroup = new THREE.Group();
         atomsGroup.name = "Atoms";
-        atomsGroup.add(...basis.coordinates.map((atomicCoordinate, atomicIndex) => {
+        basis.coordinates.forEach((atomicCoordinate, atomicIndex) => {
             const element = basis.getElementByIndex(atomicIndex);
             const sphereMesh = this.getSphereMeshObject({
                 ...this._getDefaultSettingsForElement(element, atomRadiiScale),
                 coordinate: atomicCoordinate.value,
             });
             sphereMesh.name = `${element}-${atomicIndex}`;
-            return sphereMesh;
-        }));
+            atomsGroup.add(sphereMesh);
+        });
         return atomsGroup;
     }
 
+    /**
+     * Whether to draw a bond between given elements.
+     */
+    areElementsBonded(element1, coordinate1, element2, coordinate2) {
+        return Boolean(ELEMENT_BONDS.find(b => {
+            return b.elements.includes(element1) && b.elements.includes(element2) &&
+                b.length.value && b.length.value >= Made.math.vDist(coordinate1.value, coordinate2.value);
+        }));
+    }
+
+    /**
+     * Add bonds.
+     */
+    addBonds() {
+        const basis = Made.tools.basis.repeat(this.basis, Array(3).fill(this.settings.atomRepetitions));
+        basis.coordinates.forEach((coordinate1, index1) => {
+            const element1 = basis.getElementByIndex(index1);
+            basis.coordinates.forEach((coordinate2, index2) => {
```

Comment:
This is fine for smaller structures, but nested forEach will be a performance bottleneck for bigger systems. There's no break from forEach also. Can we think of a better solution maybe?